src/SqlAst/AvgReturnTypeExtension.php

```diff
@@ -25,7 +28,23 @@ public function getReturnType(FunctionCall $expression, QueryScope $scope): ?Typ
         }
 
         $argType = $scope->getType($args[0]);
-        $numberType = $argType->toNumber();
+
+        if ($argType->isNull()->yes()) {
+            return $argType;
+        }
+        $argType = TypeCombinator::removeNull($argType);
```

Comment:
Seems unnecessary?

Reply:
Removing null is necessary for `int|null` fields to properly pass the `isInteger()` check a couple lines below. Otherwise they incorrectly fall through to the `else` branch. Would you like me to add a comment? (Or is there a better way to do this?)

Reply:
I think using CONCAT it should be possible to construct these strings, see https://github.com/staabm/phpstan-dba/blob/6b6e1ca1213fc9bcb4440a946da6b1302297edc1/src/SqlAst/ConcatReturnTypeExtension.php#L65-L76